src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -7435,22 +7504,35 @@ subroutine summarize_timestep(domain)
                              realArgs=(/global_scalar_max, latMax_global, lonMax_global/))
 
          scalar_max = -1.0e20
+         offset_1d = huge(1)
          indexMax = -1
          kMax = -1
          latMax = 0.0
          lonMax = 0.0
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(max:scalar_max)
          do iEdge = 1, nEdgesSolve
          do k = 1, nVertLevels
-            spd = sqrt(u(k,iEdge)*u(k,iEdge) + v(k,iEdge)*v(k,iEdge))
-            if (spd > scalar_max) then
-               scalar_max = spd
-               indexMax = iEdge
-               kMax = k
-               latMax = latEdge(iEdge)
-               lonMax = lonEdge(iEdge)
+            spd(k,iEdge) = sqrt(u(k,iEdge)*u(k,iEdge) + v(k,iEdge)*v(k,iEdge))
+            scalar_max = max(scalar_max, spd(k,iEdge))
+         end do
+         end do
+         !$acc end parallel
+
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(min:offset_1d)
+         do iEdge = 1, nEdgesSolve
+         do k = 1, nVertLevels
+            if (spd(k,iEdge) == scalar_max) then
+               offset_1d = min(offset_1d, (k-1) + size(u,1)*(iEdge-1))
```

Comment:
While `u` and `spd` do have the same dimensions, I'd still advocate for the use of `size(spd,1)` here (and in the code below to compute `kMax` and `indexMax`.

Reply:
Good point. Addressed by force-push from [393492e](https://github.com/MPAS-Dev/MPAS-Model/commit/393492e813271638a1c9d48a76e0b010f26ea714) to [33b0db0](https://github.com/MPAS-Dev/MPAS-Model/commit/33b0db038b47664f4c9db12345bbb95a2425a831)